backend/cmm_helpers.ml

```diff
@@ -2526,172 +2566,202 @@ let tuplify_function arity =
 
 let max_arity_optimized = 15
 
-let final_curry_function ~nlocal ~arity =
+let machtype_stored_size t =
+  Array.fold_left
+    (fun cur c ->
+      match c with
+      | Addr -> Misc.fatal_error "[Addr] cannot be stored"
+      | Val | Int -> cur + 1
+      | Float -> cur + if Arch.size_int = 4 then 2 else 1)
+    0 t
+
+let machtype_non_scanned_size t =
+  Array.fold_left
+    (fun cur c ->
+      match c with
+      | Addr -> Misc.fatal_error "[Addr] cannot be stored"
+      | Val -> cur
+      | Int -> cur + 1
+      | Float -> cur + if Arch.size_int = 4 then 2 else 1)
+    0 t
+
+let layout_machtype_for_closure t v =
+  if Array.length t > 1
+  then
+    Misc.fatal_error
+      "[layout_machtype_for_closure] currently does not support complex \
+       machtypes";
+  [Cvar v]
+
+let read_machtype_from_closure t clos base_offset dbg =
```

Comment:
The name of this function makes it sound like it's reading machtypes, let's rename it.

Reply:
We could have `read_value_from_closure`, but same problem as the above, we probably need a better term than `value`.

Reply:
`read_from_closure_given_machtype`?